File: jadx-plugins/jadx-xapk-input/src/main/java/jadx/plugins/input/xapk/XApkInputPlugin.java
package jadx.plugins.input.xapk;

import jadx.api.plugins.JadxPlugin;
import jadx.api.plugins.JadxPluginContext;
import jadx.api.plugins.JadxPluginInfo;
import jadx.api.plugins.JadxPluginInfoBuilder;

public class XApkInputPlugin implements JadxPlugin {

	private XApkLoader loader;

	@Override
	public JadxPluginInfo getPluginInfo() {
		return JadxPluginInfoBuilder.pluginId("xapk-input")
				.name("XApk Input")
				.description("Load .xapk files")
				.build();
	}

	@Override
	public void init(JadxPluginContext context) {
		loader = new XApkLoader(context);
		XApkCustomInput customInput = new XApkCustomInput(context, loader);
		context.addCodeInput(customInput);
		context.getDecompiler().addCustomResourcesLoader(customInput);
	}

	@Override
	public void unload() {
		if (loader != null) {
			loader.unload();
		}
	}
}

File: jadx-plugins/jadx-xapk-input/src/main/java/jadx/plugins/input/xapk/XapkInputPlugin.kt
package jadx.plugins.input.xapk

import jadx.api.plugins.JadxPlugin
import jadx.api.plugins.JadxPluginContext
import jadx.api.plugins.JadxPluginInfo
import jadx.api.plugins.JadxPluginInfoBuilder
import jadx.plugins.input.dex.DexInputPlugin

class XapkInputPlugin : JadxPlugin {

	override fun getPluginInfo(): JadxPluginInfo =
		JadxPluginInfoBuilder.pluginId("xapk-input")
			.name("XAPK Input")
			.description("Load .xapk files")
			.build()

	override fun init(context: JadxPluginContext) {
		val dexInputPlugin = context.plugins().getInstance(DexInputPlugin::class.java)
		context.addCodeInput(XapkCustomCodeInput(dexInputPlugin, context.zipReader))
		context.decompiler.addCustomResourcesLoader(XapkCustomResourcesLoader(context.zipReader))
	}
}
